### PR TITLE
Add Dice coefficient and upgrade Jaccard similarity

### LINKED ISSUE #149: Add Distances
As mentioned in #122 we also want to add all distance metrics. Here is a list that we should probably implement:

* [ ] Sorensen-Dice

It's very little, so adding some proposals in comments to this issue is recommended. For example, we may add kernels like linear kernel or RBF kernel and so on.

### PR DESCRIPTION
**Pull request by msluszniak**
Closes #149

I added the Dice coefficient and support for the option `axis` in Jaccard Similarity.

### CODE REVIEW

File: lib/scholar/metrics/similarity.ex [hunk 1]
```diff
@@ -6,15 +6,34 @@ defmodule Scholar.Metrics.Similarity do
   import Nx.Defn
   import Scholar.Shared
 
-  @doc ~S"""
+  general_schema = [
+    axis: [
+      type: {:custom, Scholar.Options, :axis, []},
+      doc: """
+      Axis to calculate the distance over. By default the distance
```
Comment (polvalente) [inline lib/scholar/metrics/similarity.ex:13]:
```suggestion
      Axis over which the distance will be calculated. By default the distance
```

File: lib/scholar/metrics/similarity.ex [hunk 2]
```diff
@@ -34,29 +53,59 @@ defmodule Scholar.Metrics.Similarity do
         0.25
       >
 
+      iex> x = Nx.iota({2,3})
```
Comment (polvalente) [inline lib/scholar/metrics/similarity.ex:56]: I think if you write the tensor explicitly it's easier to visualize the calculation

Comment (msluszniak) [reply lib/scholar/metrics/similarity.ex:56]: Ok

Comment (msluszniak) [reply lib/scholar/metrics/similarity.ex:56]:
```suggestion
      iex> x = Nx.tensor([[0, 1, 2], [3, 4, 5]])
```

**Review by josevalim — APPROVED** [review]

**Comment by msluszniak** [comment]
@josevalim Are branch protection rules configured correctly? Both workflows are waiting idle.

**Comment by polvalente** [comment]
@msluszniak I don't really know what happened there. Branch protection rules were fine; your actions were running correctly.
I checked out on your branch locally and did a `commit --amend + push -f` to resolve this.

**Pull request merged**